lib/src/player.dart

```diff
@@ -142,28 +125,21 @@ class Player with PlayerEventHandler implements PlayerApi {
 
   // Can be used to call methods on the platform side that return a complex
   // object that is not natively supported by the method channel.
-  Future<T?> _invokeObjectMethod<T>(
+  Future<T> _invokeObjectMethod<T>(
     String methodName,
     T Function(Map<String, dynamic>) fromJson, [
     dynamic data,
   ]) async {
-    final result = await _initializationResult;
-    if (!result) {
-      throw Exception('Error initializing player on native platform side.');
+    final String jsonString;
+    const T? tNull = null;
+    if (tNull is T) { // T is nullable
```

Comment:
this `if` condition is always `false`, or do I misunderstand something here?
(you create `tNull = null` and in the next line you check for `tNull is T` :thinking: